lib/tcpip/network_wrapper.cc

```diff
@@ -604,30 +611,34 @@ int network_socket_close(Timeout *t, SObj mallocCapability, SObj sealedSocket)
 		  }
 		  // Close the socket.  Another thread will actually clean up the
 		  // memory.
+		  int  ret         = 0;
 		  auto closeStatus = close_socket_retry(t, socket->socket);
 		  if (closeStatus == 0)
 		  {
-			  return -EINVAL;
+			  // The only reason why this would fail is internal
+			  // corruption (did someone already close the
+			  // socket?). Nothing can be done by anyone at this
+			  // stage. Don't return because we would leak
+			  // everything.
+			  ret = -ENOTRECOVERABLE;
```

Comment:
I think here we should still free the memory, but report that something went wrong?

Reply:
I'm still a bit confused.  Doesn't this leak the socket object?  We've not freed everything, but we've told the caller not to retry?